### ISSUE TITLE
Issue when multiple classes in a plugin implements
PackageMaterialProvider interface

### ISSUE THREAD

**Issue by mdaliejaz** [gocd/gocd#81]
## Description

If multiple classes in a plugin implements PackageMaterialProvider interface, the plugin tab reports that the plugin is enabled with no errors. However, the package-repo tab states that "No plugins found. To configure a package repository you must have plugin(s) installed."

**Go-server log states:**
Failed to fetch package metadata for plugin : yum
com.thoughtworks.go.plugin.infra.GoPluginFrameworkException: More than one reference found for the given Service Reference:
com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialProvider and Plugin Id yum; References:
[[com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialProvider], [com.thoughtworks.go.plugin.api.material.packagerepository.PackageMaterialProvider]]

Basically, metadata loader failed while fetching the config from the plugin (we use doOn not doOnAll). There should be some indication that the package plugin is invalid.

**Comment by arvindsv** [comment]
No activity. Stale. Closing.
